### PR TITLE
Remove SafetyNet

### PR DESCRIPTION
**Pull request by daymxn**
Per [b/333906221](https://b.corp.google.com/issues/333906221),

This drops the `SafetyNet` sdk, and its related usage throughout app-check.

Although, some of the app-check removals may not be desired, so let me know if that's the case so I can revert or change them.

### CODE REVIEW

**Review by rlazo — CHANGES_REQUESTED** [review]
needs work

File: buildSrc/src/main/java/com/google/firebase/gradle/bomgenerator/BomGeneratorTask.java [hunk 1]
```diff
@@ -105,7 +105,6 @@ public abstract class BomGeneratorTask extends DefaultTask {
           "firebase-analytics-license",
           "firebase-annotations",
           "firebase-appcheck-interop",
-          "firebase-appcheck-safetynet",
```
Comment (rlazo) [inline buildSrc/src/main/java/com/google/firebase/gradle/bomgenerator/BomGeneratorTask.java:108]: this list is based on the sdks available in maven, not in the repo. the old versions of `safetynet` are, and will be, available in maven.

Comment (daymxn) [reply buildSrc/src/main/java/com/google/firebase/gradle/bomgenerator/BomGeneratorTask.java]: fixed

**Review by rlazo — APPROVED** [review]

File: appcheck/firebase-appcheck-debug-testing/src/test/java/com/google/firebase/appcheck/debug/testing/DebugAppCheckTestHelperTest.java [hunk 2]
```diff
@@ -47,50 +46,6 @@ public void setUp() {
     initializeFirebaseApp(ApplicationProvider.getApplicationContext(), OTHER_FIREBASE_APP_NAME);
   }
 
-  @Test
```
Comment (rosalyntan) [inline appcheck/firebase-appcheck-debug-testing/src/test/java/com/google/firebase/appcheck/debug/testing/DebugAppCheckTestHelperTest.java:50]: These two tests should not be removed, as this flow of logic still needs to be verified. Can you substitute `PlayIntegrityAppCheckProviderFactory` for `SafetyNetAppCheckProviderFactory` in these tests?

File: appcheck/firebase-appcheck/CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,6 @@
 # Unreleased
+* [changed] Internal support for `SafetyNet` has been dropped, as `SafetyNet` is no longer being
```
Comment (rosalyntan) [inline appcheck/firebase-appcheck/CHANGELOG.md:2]: Maybe ".... as the `SafetyNet Attestation API` has been deprecated."

and also link to https://developer.android.com/privacy-and-security/safetynet/deprecation-timeline#safetynet_attestation_deprecation_timeline here?

**Review by rosalyntan — APPROVED** [review]

**Pull request merged**